### ISSUE TITLE
The loss is getting bigger and bigger during training

### ISSUE THREAD

**Issue by Deep-learning999** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#276]
[WARNING] Not found pre-trained parameters for model.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.1.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.2.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.2.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.3.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.3.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.4.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.4.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.5.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.5.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.6.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.6.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.7.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.7.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.8.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.8.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.9.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.9.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.10.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.10.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.11.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.11.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for cpm.align.0.weight
[WARNING] Not found pre-trained parameters for cpm.align.0.bias
[WARNING] Not found pre-trained parameters for cpm.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.0.2.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.1.2.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.2.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.2.2.weight
[WARNING] Not found pre-trained parameters for cpm.conv.0.weight
[WARNING] Not found pre-trained parameters for cpm.conv.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.2.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.2.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.1.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.pafs.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.pafs.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.pafs.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.pafs.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.1.0.bias
/usr/local/lib/python3.8/dist-packages/torch/optim/lr_scheduler.py:129: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
Iter: 100
stage1_pafs_loss:     7.6666880798339845
stage1_heatmaps_loss: 3.570530319213867
stage2_pafs_loss:     7.841795196533203
stage2_heatmaps_loss: 2.7058591079711913
Iter: 200
stage1_pafs_loss:     15.29188735961914
stage1_heatmaps_loss: 5.479909019470215
stage2_pafs_loss:     15.420273971557616
stage2_heatmaps_loss: 4.8241339111328125
Iter: 300
stage1_pafs_loss:     22.353233795166016
stage1_heatmaps_loss: 7.121137409210205
stage2_pafs_loss:     22.443800048828123
stage2_heatmaps_loss: 6.920835132598877
Iter: 400
stage1_pafs_loss:     29.957595977783203
stage1_heatmaps_loss: 9.14753137588501
stage2_pafs_loss:     30.035529327392577
stage2_heatmaps_loss: 9.002146663665771
Iter: 500
stage1_pafs_loss:     38.06448661804199
stage1_heatmaps_loss: 11.290129718780518
stage2_pafs_loss:     38.13512634277344
stage2_heatmaps_loss: 11.16331365585327
Iter: 600
stage1_pafs_loss:     44.938044204711915
stage1_heatmaps_loss: 13.224259433746338
stage2_pafs_loss:     44.98915901184082
stage2_heatmaps_loss: 13.044585227966309
Iter: 700
stage1_pafs_loss:     52.07991065979004
stage1_heatmaps_loss: 15.249440212249755
stage2_pafs_loss:     52.065321884155274
stage2_heatmaps_loss: 14.996947536468506
Iter: 800
stage1_pafs_loss:     59.267780532836916
stage1_heatmaps_loss: 17.316645069122316
stage2_pafs_loss:     59.14789474487305
stage2_heatmaps_loss: 16.928283405303954
Iter: 900
stage1_pafs_loss:     66.85546279907227
stage1_heatmaps_loss: 19.353349342346192
stage2_pafs_loss:     66.63412284851074
stage2_heatmaps_loss: 18.891154212951662

**Comment by Deep-learning999** [comment]
Iter: 1000
stage1_pafs_loss:     74.18167854309083
stage1_heatmaps_loss: 21.231216430664062
stage2_pafs_loss:     73.83959144592285
stage2_heatmaps_loss: 20.659865531921387
Iter: 1100
stage1_pafs_loss:     82.73493225097656
stage1_heatmaps_loss: 23.373098030090333
stage2_pafs_loss:     82.23949165344239
stage2_heatmaps_loss: 22.714265155792237
Iter: 1200
stage1_pafs_loss:     89.39727691650391
stage1_heatmaps_loss: 25.271691150665283
stage2_pafs_loss:     88.73115852355957
stage2_heatmaps_loss: 24.559886417388917
Iter: 1300
stage1_pafs_loss:     96.65667152404785
stage1_heatmaps_loss: 27.080880374908446
stage2_pafs_loss:     95.85724365234375
stage2_heatmaps_loss: 26.42891851425171
Iter: 1400
stage1_pafs_loss:     5.522353057861328
stage1_heatmaps_loss: 1.5060111045837403
stage2_pafs_loss:     5.480509262084961
stage2_heatmaps_loss: 1.4821682739257813
Iter: 1500
stage1_pafs_loss:     12.893040008544922
stage1_heatmaps_loss: 3.518694953918457
stage2_pafs_loss:     12.761918411254882
stage2_heatmaps_loss: 3.4662819480895997
Iter: 1600
stage1_pafs_loss:     20.37189224243164
stage1_heatmaps_loss: 5.5336927223205565
stage2_pafs_loss:     20.129829940795897
stage2_heatmaps_loss: 5.460363178253174
Iter: 1700
stage1_pafs_loss:     26.993648681640625
stage1_heatmaps_loss: 7.3423055648803714
stage2_pafs_loss:     26.650565338134765
stage2_heatmaps_loss: 7.248031311035156
Iter: 1800
stage1_pafs_loss:     34.99140800476074
stage1_heatmaps_loss: 9.37369083404541
stage2_pafs_loss:     34.43473907470703
stage2_heatmaps_loss: 9.217097358703613
Iter: 1900
stage1_pafs_loss:     42.435910034179685
stage1_heatmaps_loss: 11.361269969940185
stage2_pafs_loss:     41.68058692932129
stage2_heatmaps_loss: 11.161158351898193

**Comment by Deep-learning999** [comment]
Did you modify the layers of the model and mobilenet_sgd_68.848.pth.tar does not match this model

**Comment by Daniil-Osokin** [comment]
Hi! The output looks reasonable. Looks like the loss is bouncing. Try to increase the batch size.

**Comment by Daniil-Osokin** [comment]
Hope, it helped.
